### PR TITLE
lib: Port us away from simple deprecated Select variant

### PR DESCRIPTION
**Pull request by mvollmer**
Pixel differences: https://github.com/cockpit-project/pixel-test-reference/compare/1f17d8f6d2b39be2c9f19771004fe27c52c1bbce..baa39edeb94890093b3af6bc2e6802279e6d60eb

I think all count as bug fixes.

### CODE REVIEW

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  toggleWidth,
+  toggleProps,
+  placeholder,
+  ...props
+}: SimpleSelectProps) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const simpleSelectOptions = options.map((option) => {
+    if (typeof option === "string")
+        return <SelectOption key={option} value={option}>{option}</SelectOption>;
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:97]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    const { content, value, ...props } = option;
+    return (
+      <SelectOption value={value} key={value} {...props}>
+        {content}
+      </SelectOption>
+    );
+  });
+
+  const onToggleClick = () => {
+    onToggle && onToggle(!isOpen);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:109]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    setIsOpen(!isOpen);
+  };
+
+  const _onSelect = (_event: React.MouseEvent<Element, MouseEvent> | undefined, value: string | number | undefined) => {
+    onSelect(value);
+    onToggle && onToggle(true);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:115]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+  );
+
+  return (
+    <Select
+      isOpen={isOpen}
+      selected={selected}
+      onSelect={_onSelect}
+      onOpenChange={(isOpen) => {
+        onToggle && onToggle(isOpen);
+        setIsOpen(isOpen);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:146]: These 3 added lines are not executed by any test.

File: pkg/lib/serverTime.js [hunk 5]
```diff
@@ -581,12 +582,12 @@ function ChangeSystimeBody({ state, errors, change }) {
...
-                        selections={mode} onSelect={(event, value) => { setModeOpen(false); change("mode", value) }}
-                        menuAppendTo="parent">
-                    { mode_options }
-                </Select>
+                <SimpleSelect
+                    id="change_systime"
+                    placeholder={_("Mode")}
+                    options={mode_options}
+                    selected={mode}
+                    onSelect={value => change("mode", value)} />
```
Comment (cockpituous) [inline pkg/lib/serverTime.js:590]: This added line is not executed by any test.

File: pkg/metrics/metrics.jsx [hunk 6]
```diff
@@ -1632,7 +1633,7 @@ class MetricsHistory extends React.Component {
         this.setState({ isDatepickerOpened: isOpen });
     }
 
-    handleSelect(e, sel) {
+    handleSelect(sel) {
```
Comment (cockpituous) [inline pkg/metrics/metrics.jsx:1636]: This added line is not executed by any test.

File: pkg/systemd/logs.jsx [hunk 7]
```diff
@@ -167,19 +173,21 @@ export const LogsPage = () => {
         updateUrl(Object.assign(getOptionsFromTextInput(value)));
     };
 
-    const onTimeFilterChange = (newTimeFilter) => {
+    const onTimeFilterChange = (value) => {
```
Comment (cockpituous) [inline pkg/systemd/logs.jsx:176]: This added line is not executed by any test.

File: pkg/systemd/logs.jsx [hunk 8]
```diff
@@ -167,19 +173,21 @@
...
         setUpdateIdentifiersList(true);
 
-        if (newTimeFilter.key == 'boot' && newTimeFilter.value !== "0") // Don't follow if specific boot is picked
+        const newTimeFilter = timeFilterParams[value];
```
Comment (cockpituous) [inline pkg/systemd/logs.jsx:179]: This added line is not executed by any test.

File: pkg/systemd/logs.jsx [hunk 9]
```diff
@@ -167,19 +173,21 @@
...
+
+        if (newTimeFilter.param == 'boot' && newTimeFilter.value !== "0") // Don't follow if specific boot is picked
```
Comment (cockpituous) [inline pkg/systemd/logs.jsx:181]: This added line is not executed by any test.

File: pkg/systemd/logs.jsx [hunk 10]
```diff
@@ -167,19 +173,21 @@
...
             setDataFollowing(false);
-        else if (options.boot && options.boot !== "0" && newTimeFilter.key !== "boot") // Start following is specific boot is removed
+        else if (options.boot && options.boot !== "0" && newTimeFilter.param !== "boot") // Start following is specific boot is removed
```
Comment (cockpituous) [inline pkg/systemd/logs.jsx:183]: This added line is not executed by any test.

File: pkg/systemd/logs.jsx [hunk 11]
```diff
@@ -167,19 +173,21 @@
...
             setDataFollowing(true);
 
         // Remove all parameters which can be set up using filters
         delete options.boot;
         delete options.since;
 
-        cockpit.location.go([], Object.assign(options, { [newTimeFilter.key]: newTimeFilter.value }));
+        cockpit.location.go([], Object.assign(options, { [newTimeFilter.param]: newTimeFilter.value }));
```
Comment (cockpituous) [inline pkg/systemd/logs.jsx:190]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+    const { content, value, ...props } = option;
+    return (
+      <SelectOption value={value} key={value} {...props}>
+        {content}
+      </SelectOption>
+    );
+  });
+
+  const onToggleClick = () => {
+    onToggle && onToggle(!isOpen);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:127]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+    setIsOpen(!isOpen);
+  };
+
+  const _onSelect = (_event: React.MouseEvent<Element, MouseEvent> | undefined, value: string | number | undefined) => {
+    if (value)
+      onSelect(value);
+    onToggle && onToggle(true);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:134]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 14]
```diff
@@ -0,0 +1,184 @@
...
+    setIsOpen(false);
+  };
+
+  let content: React.ReactNode = placeholder;
+  if (toggleContent)
+    content = toggleContent;
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:140]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 15]
```diff
@@ -0,0 +1,184 @@
...
+  );
+
+  return (
+    <Select
+      isOpen={isOpen}
+      selected={selected}
+      onSelect={_onSelect}
+      onOpenChange={(isOpen) => {
+        onToggle && onToggle(isOpen);
+        setIsOpen(isOpen);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:168]: These 3 added lines are not executed by any test.

File: pkg/systemd/logs.jsx [hunk 16]
```diff
@@ -253,7 +253,7 @@ export const LogsPage = () => {
                                                 isDisabled={options.boot && options.boot !== "0"}
                                                 onClick={() => {
                                                     // Reset time filter if following mode is now selected but we are on a specific boot
-                                                    if (!dataFollowing && timeFilter && timeFilter.key == "boot" && timeFilter.value !== "0") {
+                                                    if (!dataFollowing && timeFilterParam && timeFilterParam.param == "boot" && timeFilterParam.value !== "0") {
```
Comment (cockpituous) [inline pkg/systemd/logs.jsx:256]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx (hunk 12 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:127]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx (hunk 13 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:134]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx (hunk 14 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:140]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx (hunk 15 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:168]: These 3 added lines are not executed by any test.

File: pkg/systemd/logs.jsx (hunk 16 above)
Comment (cockpituous) [inline pkg/systemd/logs.jsx:256]: This added line is not executed by any test.

**Comment by mvollmer** [comment]
@garrett, how should normal dropdown selects look?

Like this (a):

image

Or like this (b):

image

FormSelect and the deprecated Select that we are still using in many places look like (a).  The new simple Select introduced in this PR would look like (b) because of our fix for https://github.com/patternfly/patternfly/issues/6632 in patternfly-5-overrides.scss. With upstream PF5 styling, the new simple Selects would also look like (a).

Since we have traditionally been okay with how FormSelect and the deprecated Select look, I assume for now that the new simple Select should continue to look like them, i.e. like (a).  

Consequently, I have narrowed the scope of the "double spacing" bug fix to the dropdowns in the Shell masthead. I think that's the only place where we want it (assuming (a) is the desired look for select dropdowns). Without the bug fix, the masthead dropdowns look uncentered since the large icon on their left doesn't have a margin, while the dropdown icon on the right has one.

**Comment by mvollmer** [comment]
The pixel tests and `git grep` found 2 dropdowns that are affected by our current "double spacing" bug fix: zoom controls and the "Add member" menu of bonds/bridges/etc.

Consequently, they currently look like (b) above, and thus look different from other dropdowns.  This PR will change them to look like (a) as a side effect.

**Comment by mvollmer** [comment]
If we decide that dropdown should look like (b) everywhere (which I wouldn't object to at all), then we should do that in one step and change the looks of the deprecated Select and FormSelect in a separate PR and block this one on it.

**Comment by mvollmer** [comment]
@garrett, now about the background color... The non-deprecated MenuToggle that this PR uses for simple Select dropdown has a transparent background in light mode, while the deprecated Select has regular white. This makes a difference when the page background is not also white, like in toolbars:

With deprecated Select:

image

With non-deprecated MenuToggle:

image

Note that the "Filters" SearchInput input group has a white background but the "Pause" button is transparent. As far as I can tell, this is the default PF behavior.

Which should it be?

As with the icon margins, I have made it so that this PR keeps the current look.

File: pkg/lib/cockpit-components-simple-select.tsx (hunk 12 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:127]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx (hunk 13 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:134]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx (hunk 14 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:140]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx (hunk 15 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:168]: These 3 added lines are not executed by any test.

File: pkg/systemd/logs.jsx (hunk 16 above)
Comment (cockpituous) [inline pkg/systemd/logs.jsx:256]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx (hunk 13 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:134]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx (hunk 14 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:140]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx (hunk 15 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:168]: These 3 added lines are not executed by any test.

File: pkg/systemd/logs.jsx (hunk 16 above)
Comment (cockpituous) [inline pkg/systemd/logs.jsx:256]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 17]
```diff
@@ -42,24 +51,38 @@ import {
...
 
-export interface SimpleSelectOption extends Omit<SelectOptionProps, 'content'> {
+export interface SimpleSelectDividerOption {
+  decorator: "divider";
+
+  key: string | number;
+};
+
+export interface SimpleSelectMenuOption extends Omit<SelectOptionProps, 'content'> {
+  decorator?: undefined;
```
Comment (martinpitt) [inline pkg/lib/cockpit-components-simple-select.tsx:66]: OOI, why is that necessary? It means that the only possible value is `undefined`, right? How is that useful?

**Comment by garrett** [comment]
b is definitely incorrect; a looks more correct at a glance

the widgets should all have a white background there, including the secondary button; they need contrast... the text on grey isn't enough contrast, and additionally the grey on the widgets make them look disabled

File: pkg/systemd/logs.jsx [hunk 18]
```diff
@@ -55,36 +55,45 @@ import "./logs.scss";
...
-    { key: "since", value: "-24hours", toString: () => _("Last 24 hours"), default: true },
-    { key: "since", value: "-7days", toString: () => _("Last 7 days") },
+    { value: "current", content: _("Current boot") },
+    { value: "previous", content: _("Previous boot") },
+    { value: "last24h", content: _("Last 24 hours") },
+    { value: "last7d", content: _("Last 7 days") },
 ];
 
+const timeFilterParams = {
+    current: { param: "boot", value: 0 },
```
Comment (martinpitt) [inline pkg/systemd/logs.jsx:65]: Could we somehow allow "any additional junk" in the type declaration of MenuItem, so that we can continue to add these extra fields right in their declaration? and they are passed through event handlers opaquely?

File: pkg/lib/serverTime.js [hunk 19]
```diff
@@ -554,18 +552,19 @@ function ChangeSystimeBody({ state, errors, change }) {
     );
 
     const mode_options = [
-        <SelectOption key="manual_time" value="manual_time">{_("Manually")}</SelectOption>,
-        <SelectOption key="ntp_time" value="ntp_time" isDisabled={!ntp_supported}>{_("Automatically using NTP")}</SelectOption>
+        { value: "manual_time", content: _("Manually") },
+        { value: "ntp_time", content: _("Automatically using NTP") },
```
Comment (martinpitt) [inline pkg/lib/serverTime.js:556]: This lost the isDisabled. It ought to be possible to specify that right here, as you do that below for ntp_time_custom

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks! The only actual bug here (I think) is the accidentally dropped `isDisabled`; plus two questions.

File: pkg/lib/cockpit-components-simple-select.tsx (hunk 17 above)
Comment (mvollmer) [reply pkg/lib/cockpit-components-simple-select.tsx:66]: This is to teach TypeScript how to distinguish between SimpleSelectMenuOption and SimpleSelectDividerOption: https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions

File: pkg/lib/serverTime.js (hunk 19 above)
Comment (mvollmer) [reply pkg/lib/serverTime.js:556]: Ouch, thanks!

File: pkg/systemd/logs.jsx (hunk 18 above)
Comment (mvollmer) [reply pkg/systemd/logs.jsx:65]: What about allowing
```
{ 
  content: _("Current boot"),
  value: { key: "boot", value: 0 },
}
```

and we use `content` as the React key?

Comment (martinpitt) [reply pkg/systemd/logs.jsx:65]: FTR, I'm completely happy with the `value`s here. Restricting them to strings (or perhaps also numbers) keeps things easy and non-ambiguous. I was just asking if we could merge` timeFilterParams` into `timeFilterOptions` by just adding extra `param` and `systemd_value` (or similar) keys.

Both value and content are fine for the key, but I think `value` is a bit cleaner (i.e. the status quo)

Comment (martinpitt) [reply pkg/systemd/logs.jsx:65]: In other words, can we declare `SimpleSelectMenuOption` as content: string, value: string | number, plus "any additional stuff" like `[string]: unknown`? 

If that's difficult/impractical then ignore me. I was just wondering if object types can be partially specified and partially opaque.

Comment (mvollmer) [reply pkg/systemd/logs.jsx]: Done. `content` can be any React.ReactNode so it can't always be used as a key, so we fall back to the ... index. :-/ But a explicit key in SimpleSelectOption is also supported.

File: pkg/lib/serverTime.js (hunk 19 above)
Comment (mvollmer) [reply pkg/lib/serverTime.js:556]: Fixed.

File: pkg/lib/cockpit-components-simple-select.tsx (hunk 17 above)
Comment (mvollmer) [reply pkg/lib/cockpit-components-simple-select.tsx:69]: (I guess `decorator?: never` might be more ideomatic.)

**Comment by mvollmer** [comment]
> b is definitely incorrect; a looks more correct at a glance

Ok, thanks! Then this PR is already correct.
 
> the widgets should all have a white background there, including the secondary button; they need contrast... the text on grey isn't enough contrast, and additionally the grey on the widgets make them look disabled

Right, the this PR is also correct re the background.

I'll fix the button in a separate PR.

Thanks!

File: pkg/systemd/logs.jsx (hunk 18 above)
Comment (mvollmer) [reply pkg/systemd/logs.jsx]: > In other words, can we declare SimpleSelectMenuOption as content: string, value: string | number, plus "any additional stuff" like [string]: unknown?

Hmm, we could, but I think it's not a good idea. It already has all the options from SelectOption, which inherits from MenuItem, and I think the risk of collision is just too big. For example, both `key` and `value` are already taken.

The old code also didn't do this. It didn't have 
```
<SelectOption key='boot' value='-1'>Previous boot</SelectOption>
```
in it, but constructed it from timeFilterOptions.

Comment (martinpitt) [reply pkg/systemd/logs.jsx]: I see. I clearly don't understand this enough, and it looks fine - I was just curious.  So let's just keep this. Thanks!

File: pkg/systemd/logs.jsx [hunk 20]
```diff
@@ -55,36 +55,37 @@ import "./logs.scss";
...
 
 const timeFilterOptions = [
-    { key: "boot", value: 0, toString: () => _("Current boot"), },
-    { key: "boot", value: "-1", toString: () => _("Previous boot") },
-    { key: "since", value: "-24hours", toString: () => _("Last 24 hours"), default: true },
-    { key: "since", value: "-7days", toString: () => _("Last 7 days") },
+    { value: { key: "boot", value: 0 }, content: _("Current boot") },
+    { value: { key: "boot", value: "-1" }, content: _("Previous boot") },
+    { value: { key: "since", value: "-24hours", default: true }, content: _("Last 24 hours") },
+    { value: { key: "since", value: "-7days" }, content: _("Last 7 days") },
```
Comment (martinpitt) [inline pkg/systemd/logs.jsx:61]: How are these not duplicate React keys?

I know the old code had this as well, and maybe that's just misleading -- these may not actually be React keys, but a custom value that we use to construct the `journalctl` argv? But then, the [`value` is being used as the React key](https://github.com/cockpit-project/cockpit/pull/21356/commits/0ac10078c22d4218ff170f121f36c4fd9f542e86#diff-cde3c8035b0caedbd25d7c632e9be44b71fca3cd18e6905221761695d3a73051R101)  And that should be a string, not an object. How does this work?

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks! One more raised eyebrow.

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 21]
```diff
@@ -0,0 +1,187 @@
...
+  ...props
+}: SimpleSelectProps) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const simpleSelectOptions = options.map((option, index) => {
+    if (option.decorator == "divider")
+      return <Divider key={option.key} component="li" />;
+    const { content, value, key, ...props } = option;
+    return (
+      <SelectOption value={value} key={key || (typeof content == "string" ? content : index)} {...props}>
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:123]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 22]
```diff
@@ -0,0 +1,187 @@
...
+        {content}
+      </SelectOption>
+    );
+  });
+
+  const onToggleClick = () => {
+    onToggle && onToggle(!isOpen);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:130]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 23]
```diff
@@ -0,0 +1,187 @@
...
+    setIsOpen(!isOpen);
+  };
+
+  const _onSelect = (_event: React.MouseEvent<Element, MouseEvent> | undefined, value: unknown) => {
+    if (value)
+      onSelect(value);
+    onToggle && onToggle(true);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:137]: This added line is not executed by any test.

**Comment by martinpitt** [comment]
[This pixel change](https://cockpit-logs.us-east-1.linodeobjects.com/pull-21356-07bf0df3-20241210-125322-fedora-41/pixeldiff.html#TestNetworkingBasic-testBasic-network-main) feels like a regression to me, but if that's hard to fix, I don't consider it a blocker. [This pixel change](https://cockpit-logs.us-east-1.linodeobjects.com/pull-21356-07bf0df3-20241210-125322-fedora-41/pixeldiff.html#TestJournal-testBasic-journal-filters-panel-dark) is a bug fix and can be accepted into the reference.

File: pkg/systemd/logs.jsx [hunk 24]
```diff
@@ -55,36 +55,37 @@ import "./logs.scss";
...
 
 const timeFilterOptions = [
-    { key: "boot", value: 0, toString: () => _("Current boot"), },
-    { key: "boot", value: "-1", toString: () => _("Previous boot") },
-    { key: "since", value: "-24hours", toString: () => _("Last 24 hours"), default: true },
-    { key: "since", value: "-7days", toString: () => _("Last 7 days") },
+    { key: 1, value: { key: "boot", value: 0 }, content: _("Current boot") },
+    { key: 2, value: { key: "boot", value: "-1" }, content: _("Previous boot") },
+    { key: 3, value: { key: "since", value: "-24hours", default: true }, content: _("Last 24 hours") },
+    { key: 4, value: { key: "since", value: "-7days" }, content: _("Last 7 days") },
```
Comment (martinpitt) [inline pkg/systemd/logs.jsx:61]: Thanks, that avois the confusion around what `key` is. (A bit weird to use numbers which don't mean anything, but that's fine).

**Review by martinpitt — APPROVED** [review]
Thanks! As I wrote above, the pixel refs need to be updated. I'm ok with that spacing regression becoming a follow-up, unless you already know how to fix it and it's easy.

Otherwise, code looks nice to me!

**Review by martinpitt — APPROVED** [review]
Thanks! You removed no-test after force-pushing, so nothing ran yet. I triggered the tests.

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 25]
```diff
@@ -0,0 +1,191 @@
...
+    const { content, value, key, ...props } = option;
+    return (
+      <SelectOption value={value} key={key !== undefined ? key : `${value}`} {...props}>
+        {content}
+      </SelectOption>
+    );
+  });
+
+  const onToggleClick = () => {
+    onToggle && onToggle(!isOpen);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:134]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 26]
```diff
@@ -0,0 +1,191 @@
...
+    setIsOpen(!isOpen);
+  };
+
+  const _onSelect = (_event: React.MouseEvent<Element, MouseEvent> | undefined, value: unknown) => {
+    if (value)
+      onSelect(value);
+    onToggle && onToggle(true);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:141]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 27]
```diff
@@ -0,0 +1,191 @@
...
+    setIsOpen(false);
+  };
+
+  let content: React.ReactNode = placeholder;
+  if (toggleContent)
+    content = toggleContent;
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:147]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-simple-select.tsx [hunk 28]
```diff
@@ -0,0 +1,191 @@
...
+  );
+
+  return (
+    <Select
+      isOpen={isOpen}
+      selected={selected}
+      onSelect={_onSelect}
+      onOpenChange={(isOpen) => {
+        onToggle && onToggle(isOpen);
+        setIsOpen(isOpen);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-simple-select.tsx:175]: These 3 added lines are not executed by any test.
